### ISSUE TITLE
[torrentz2] Exception (torrentz2): The requested address is not valid in this context: An error occurred while sending the request (Test)

### ISSUE THREAD

**Issue by hjmuller** [Jackett/Jackett#4040]
**Please use the search bar** at the top of the page and make sure you are not creating an already submitted issue. Duplicating issues makes it more difficult for everyone to follow. Your issue may have already been solved in the past as well.

Provide a description of the feature request or bug, the more details the better.

If you are experiencing an issue with a tracker, then:
1. Use your browser to check you can access the site directly, and if a login is required, check you can login and that you do not have any outstanding account issues. - **can access directly**
2. If you haven't already, try upgrading to the latest version of Jackett, your issue may have already been resolved. - **already done**

3. If it is still not working for you, then a **full enhanced log must be included**. Instructions for obtaining logs are here: https://github.com/Jackett/Jackett#troubleshooting - **see
[log.txt](https://github.com/Jackett/Jackett/files/2499836/log.txt)**

**Jackett version**:  **0.10.304.0**
**Mono version** (if not using Windows):  5.16.0.179
Debian GNU/Linux 9 \n \l
-------------------------------

**Comment by garfield69** [comment]
on the same system that you are running Jackett, (presumably windows since you did not specify a mono version), open a browser and try https://torrentz2.eu/searchA?f=&safe=1
what do you get?
I don't have any problems using Jackett and torrentz2 on my win10 so i'm thinking you are being blocked by your ISP. 
the browser test above will confirm this or eliminate it from the list of possible causes.
also please confirm your platform. Thanks.

**Comment by hjmuller** [comment]
I have no issues accessing the site from the system itself.

`Debian GNU/Linux 9 \n \l`

```
Mono JIT compiler version 5.16.0.179 (tarball Thu Oct  4 10:24:32 UTC 2018)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
        TLS:           __thread
        SIGSEGV:       altstack
        Notifications: epoll
        Architecture:  amd64
        Disabled:      none
        Misc:          softdebug
        Interpreter:   yes
        LLVM:          yes(3.6.0svn-mono-/)
        GC:            sgen (concurrent by default)

```

**Comment by garfield69** [comment]
@kaso17  @flightlevel  `The requested address is not valid in this context` smacks of socket issues doesn't it? 
Could you take a look at this ticket, please?  Thanks.

**Comment by kaso17** [comment]
Please provide a full enhanced log. Instructions for obtaining logs are here: https://github.com/Jackett/Jackett#troubleshooting

**Comment by garfield69** [comment]
OP provided the log in a re-edit of the first post.  I've copied the link here [log.txt](https://github.com/Jackett/Jackett/files/2497752/log.txt)

**Comment by kaso17** [comment]
On the jackett side everything looks fine.
I would suspect some issue with the DNS (returning invalid IPs) or broken IPv6 connectivity.

**Comment by hjmuller** [comment]
I believe I resolved it. I have pihole running and managing DNS. Once I whitelisted the domain it works fine. Thanks!

**Comment by restalfep** [comment]
Why did you have to whitelist? I am also using pi.hole but don't see any blocked DNS records for Torrentz2. It works fine when I go direct but hasn't worked for awhile now in Jackett for me. I always get: Exception (torrentz2): A task was canceled.: A task was canceled. Assuming it is probably related to your issue and pihole also.

**Comment by hjmuller** [comment]
> 
> 
> What exact domain did you have to whitelist? I am also using pi.hole but don't see any blocked DNS records for Torrentz2. It works fine when I go direct but hasn't worked for awhile now in Jackett for me. I always get: Exception (torrentz2): A task was canceled.: A task was canceled. Assuming it is probably related to your issue and pihole also.

I whitelisted torrentz2.eu. The strange thing was that when I accessed the site in a browser I had no issues, but the site did not work in Jackett. The URL came up as blocked in my pihole, and when I whitelisted, I had no issues in Jackett.

**Comment by restalfep** [comment]
That is strange, my pihole isn't blocking it at all and I tried whitelisting it now also just incase but makes no difference. Torrentz2 works fine in my browser but in Jackett it just keeps trying for awhile it seems then says: Exception (torrentz2): A task was canceled.: A task was canceled.

Very strange.

**Comment by MLSci** [comment]
> That is strange, my pihole isn't blocking it at all and I tried whitelisting it now also just incase but makes no difference. Torrentz2 works fine in my browser but in Jackett it just keeps trying for awhile it seems then says: Exception (torrentz2): A task was canceled.: A task was canceled.
> 
> Very strange.

Were you able to find a solution?
